AssetRipper.Fundamentals/Configuration/CoreConfiguration.cs

```diff
@@ -3,12 +3,39 @@
 using AssetRipper.Core.Logging;
 using AssetRipper.Core.Parser.Files;
 using AssetRipper.Core.Utils;
+using System.Collections.Generic;
 using System.IO;
 
 namespace AssetRipper.Core.Configuration
 {
 	public class CoreConfiguration
 	{
+		public Dictionary<Type, object> settings = new Dictionary<Type, object>();
```

Comment:
This uses boxing, which I oppose almost as strongly as reflection.

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/boxing-and-unboxing

Reply:
If we're to avoid using code duplication, I see only these two paths, either we use reflection (even if we do codegen, it would have to use reflection to get the member names), or we use a centralised store object to house all settings. I guess we could do some kind of overloading to avoid boxing though, but that feels super messy from a maintenance perspective.